artemis-server/src/main/java/org/apache/activemq/artemis/core/paging/impl/Page.java

```diff
@@ -223,7 +223,7 @@ private int read(StorageManager storage, ActiveMQBuffer fileBuffer, List<PagedMe
 
    public synchronized void write(final PagedMessage message) throws Exception {
       if (!file.isOpen()) {
-         return;
+         throw new IllegalStateException("can't write to closed file " + file);
```

Comment:
IllegalSTateException is not much better on this context.


Do you have a case where this is happening? 

We shouldn't have a case where the file is open and still trying to write it.


Where do you see it?

Reply:
This is from https://github.com/hornetq/hornetq/pull/2122 
(long forgotten by me. But as the pr noted that this causes random failure in PagingTest#testExpireLargeMessageOnPaging()).

Reply:
ok.. I think you should throw a proper exception, extending ActiveMQException:

ActiveMQIOErrorException would be perfect there.

Reply:
ok will do. thx.